File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/search_results_group_by_collection.py
# coding: utf-8
# Copyright (c) 2016, 2026, Oracle and/or its affiliates.  All rights reserved.
# This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.


from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class SearchResultsGroupByCollection(object):
    """
    Grouped list of objects by type.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new SearchResultsGroupByCollection object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param count:
            The value to assign to the count property of this SearchResultsGroupByCollection.
        :type count: int

        :param query:
            The value to assign to the query property of this SearchResultsGroupByCollection.
        :type query: str

        :param items:
            The value to assign to the items property of this SearchResultsGroupByCollection.
        :type items: dict(str, list[ObjectSearchSummary])

        :param aggregations:
            The value to assign to the aggregations property of this SearchResultsGroupByCollection.
        :type aggregations: dict(str, list[SearchObjectType])

        """
        self.swagger_types = {
            'count': 'int',
            'query': 'str',
            'items': 'dict(str, list[ObjectSearchSummary])',
            'aggregations': 'dict(str, list[SearchObjectType])'
        }

        self.attribute_map = {
            'count': 'count',
            'query': 'query',
            'items': 'items',
            'aggregations': 'aggregations'
        }

        self._count = None
        self._query = None
        self._items = None
        self._aggregations = None

    @property
    def count(self):
        """
        **[Required]** Gets the count of this SearchResultsGroupByCollection.
        Total number of hits


        :return: The count of this SearchResultsGroupByCollection.
        :rtype: int
        """
        return self._count

    @count.setter
    def count(self, count):
        """
        Sets the count of this SearchResultsGroupByCollection.
        Total number of hits


        :param count: The count of this SearchResultsGroupByCollection.
        :type: int
        """
        self._count = count

    @property
    def query(self):
        """
        **[Required]** Gets the query of this SearchResultsGroupByCollection.
        User search query


        :return: The query of this SearchResultsGroupByCollection.
        :rtype: str
        """
        return self._query

    @query.setter
    def query(self, query):
        """
        Sets the query of this SearchResultsGroupByCollection.
        User search query


        :param query: The query of this SearchResultsGroupByCollection.
        :type: str
        """
        self._query = query

    @property
    def items(self):
        """
        **[Required]** Gets the items of this SearchResultsGroupByCollection.
        Objects grouped by their type. Each key is a type (e.g., \"database\", \"catalog\").


        :return: The items of this SearchResultsGroupByCollection.
        :rtype: dict(str, list[ObjectSearchSummary])
        """
        return self._items

    @items.setter
    def items(self, items):
        """
        Sets the items of this SearchResultsGroupByCollection.
        Objects grouped by their type. Each key is a type (e.g., \"database\", \"catalog\").


        :param items: The items of this SearchResultsGroupByCollection.
        :type: dict(str, list[ObjectSearchSummary])
        """
        self._items = items

    @property
    def aggregations(self):
        """
        **[Required]** Gets the aggregations of this SearchResultsGroupByCollection.
        Grouped doc name with count.


        :return: The aggregations of this SearchResultsGroupByCollection.
        :rtype: dict(str, list[SearchObjectType])
        """
        return self._aggregations

    @aggregations.setter
    def aggregations(self, aggregations):
        """
        Sets the aggregations of this SearchResultsGroupByCollection.
        Grouped doc name with count.


        :param aggregations: The aggregations of this SearchResultsGroupByCollection.
        :type: dict(str, list[SearchObjectType])
        """
        self._aggregations = aggregations

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client-2x/src/main/java/com/oracle/aidataplatform/dp/model/SearchResultsGroupByCollection.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;


/**
 * Grouped list of objects by type.
**/
@javax.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=SearchResultsGroupByCollection.Builder.class)

public final class SearchResultsGroupByCollection  {
    @Deprecated
    @java.beans.ConstructorProperties({"count", "query", "items", "aggregations"})
    public SearchResultsGroupByCollection(Integer count, String query, java.util.Map<String, java.util.List<ObjectSearchSummary>> items, java.util.Map<String, java.util.List<SearchObjectType>> aggregations) {
        super();
        this.count = count;
        this.query = query;
        this.items = items;
        this.aggregations = aggregations;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
            /**
     * Total number of hits
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("count")
        private Integer count;

                /**
         * Total number of hits
         * @param count the value to set
         * @return this builder
         **/
        

        public Builder count(Integer count) {
        this.count = count;
        return this;
        }
            /**
     * User search query
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("query")
        private String query;

                /**
         * User search query
         * @param query the value to set
         * @return this builder
         **/
        

        public Builder query(String query) {
        this.query = query;
        return this;
        }
            /**
     * Objects grouped by their type. Each key is a type (e.g., "database", "catalog").
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("items")
        private java.util.Map<String, java.util.List<ObjectSearchSummary>> items;

                /**
         * Objects grouped by their type. Each key is a type (e.g., "database", "catalog").
         * @param items the value to set
         * @return this builder
         **/
        

        public Builder items(java.util.Map<String, java.util.List<ObjectSearchSummary>> items) {
        this.items = items;
        return this;
        }
            /**
     * Grouped doc name with count.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("aggregations")
        private java.util.Map<String, java.util.List<SearchObjectType>> aggregations;

                /**
         * Grouped doc name with count.
         * @param aggregations the value to set
         * @return this builder
         **/
        

        public Builder aggregations(java.util.Map<String, java.util.List<SearchObjectType>> aggregations) {
        this.aggregations = aggregations;
        return this;
        }


        public SearchResultsGroupByCollection build() {
            SearchResultsGroupByCollection model = new SearchResultsGroupByCollection(this.count
                    , this.query
                    , this.items
                    , this.aggregations);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(SearchResultsGroupByCollection model) {
                this.count(model.getCount());
    this.query(model.getQuery());
    this.items(model.getItems());
    this.aggregations(model.getAggregations());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * Total number of hits
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("count")
    private final Integer count;

        /**
     * Total number of hits
     * @return the value
     **/
    
    public Integer getCount() {
        return count;
    }


        /**
     * User search query
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("query")
    private final String query;

        /**
     * User search query
     * @return the value
     **/
    
    public String getQuery() {
        return query;
    }


        /**
     * Objects grouped by their type. Each key is a type (e.g., "database", "catalog").
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("items")
    private final java.util.Map<String, java.util.List<ObjectSearchSummary>> items;

        /**
     * Objects grouped by their type. Each key is a type (e.g., "database", "catalog").
     * @return the value
     **/
    
    public java.util.Map<String, java.util.List<ObjectSearchSummary>> getItems() {
        return items;
    }


        /**
     * Grouped doc name with count.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("aggregations")
    private final java.util.Map<String, java.util.List<SearchObjectType>> aggregations;

        /**
     * Grouped doc name with count.
     * @return the value
     **/
    
    public java.util.Map<String, java.util.List<SearchObjectType>> getAggregations() {
        return aggregations;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("SearchResultsGroupByCollection(");
        sb.append("count=").append(String.valueOf(this.count));
        sb.append(", query=").append(String.valueOf(this.query));
        sb.append(", items=").append(String.valueOf(this.items));
        sb.append(", aggregations=").append(String.valueOf(this.aggregations));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof SearchResultsGroupByCollection)) {
            return false;
        }

        SearchResultsGroupByCollection other = (SearchResultsGroupByCollection) o;
        return java.util.Objects.equals(this.count, other.count) &&
            java.util.Objects.equals(this.query, other.query) &&
            java.util.Objects.equals(this.items, other.items) &&
            java.util.Objects.equals(this.aggregations, other.aggregations);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.count == null ? 43 : this.count.hashCode());
        result = (result * PRIME) + (this.query == null ? 43 : this.query.hashCode());
        result = (result * PRIME) + (this.items == null ? 43 : this.items.hashCode());
        result = (result * PRIME) + (this.aggregations == null ? 43 : this.aggregations.hashCode());
        return result;
    }


}
